From: Keith Busch <keith.busch@intel.com>
Date: Fri, 27 Oct 2017 13:51:22 -0600
Subject: nvme: Fix setting logical block format when revalidating
Patch-mainline: v4.14-rc8
Git-commit: 5e0fab57fb3e0e553758067a15eefdc796ef0a76
References: bsc#1079313

Revalidating the disk needs to set the logical block format and capacity,
otherwise it can't figure out if the users modified anything about
the namespace.

Fixes: cdbff4f26bd9 ("nvme: remove nvme_revalidate_ns")

Reviewed-by: Christoph Hellwig <hch@lst.de>
Reviewed-by: Sagi Grimberg <sagi@grimberg.me>
Signed-off-by: Keith Busch <keith.busch@intel.com>
Signed-off-by: Jens Axboe <axboe@kernel.dk>
Acked-by: Johannes Thumshirn <jthumshirn@suse.de>
---
 drivers/nvme/host/core.c |    1 +
 1 file changed, 1 insertion(+)

--- a/drivers/nvme/host/core.c
+++ b/drivers/nvme/host/core.c
@@ -1086,6 +1086,7 @@ static int nvme_revalidate_disk(struct g
 		goto out;
 	}
 
+	__nvme_revalidate_disk(disk, id);
 	nvme_report_ns_ids(ctrl, ns->ns_id, id, ns->eui, ns->nguid, ns->uuid);
 out:
 	kfree(id);
